tekton/ci/tekton-image-build.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: tekton-image-build
+  namespace: tektonci
+spec:
+  params:
+    - name: gitCloneDepth
+      description: Number of commits in the change + 1
+    - name: fileFilterRegex
+      description: Names regex to be matched in the list of modified files
+    - name: checkName
+      description: The name of the GitHub check that this pipeline is used for
+    - name: imageName
+      description: The imageName to build
+    - name: imagePrefix
+      description: The prefix for the image reference to build
+      default: gcr.io/tekton-nightly/dogfooding/
```

Comment:
Wasn't sure what image prefix to use :angel: 
@afrittoli is the serviceAccount allowed to push there ? (I didn't want to pollute `gcr.o/tekton-release/…` :stuck_out_tongue_closed_eyes: )

Reply:
Good point, the service account used for CI jobs is not allowed to pushed anywhere right now.
`tekton-nightly` is probably better then `tekton-release`, we could add the relevant secrets but you may have to mount them explicitly in the task similar to what we do in the release job.

Perhaps we could use   `gcr.io/tekton-nightly/dogfooding/ci`